src/Entity/ListBuilder/AppListBuilder.php

```diff
@@ -84,19 +92,25 @@ class AppListBuilder extends EdgeEntityListBuilder {
    *   The request stack object.
    * @param \Drupal\Component\Datetime\TimeInterface $time
    *   The time service.
-   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   * @param \Drupal\Core\Config\ConfigFactoryInterface|null $config_factory
    *   The config factory.
+   * @param \Drupal\apigee_edge\Entity\AppWarningsCheckerInterface $app_warnings_checker
+   *   The app warnings checker service.
    */
-  public function __construct(EntityTypeInterface $entity_type, EntityTypeManagerInterface $entity_type_manager, RendererInterface $renderer, RequestStack $request_stack, TimeInterface $time, ConfigFactoryInterface $config_factory = NULL) {
+  public function __construct(EntityTypeInterface $entity_type, EntityTypeManagerInterface $entity_type_manager, RendererInterface $renderer, RequestStack $request_stack, TimeInterface $time, ConfigFactoryInterface $config_factory = NULL, AppWarningsCheckerInterface $app_warnings_checker = NULL) {
     if (!$config_factory) {
       $config_factory = \Drupal::service('config.factory');
     }
+    if (!$app_warnings_checker) {
+      $app_warnings_checker = \Drupal::service('apigee_edge.entity.app_warnings_checker');
```

Comment:
An error should be also triggered if a required parameter was missing and a fallback solution had to be applied
See: https://github.com/drupal/core/blob/8.8.x/lib/Drupal/Core/Entity/ContentEntityStorageBase.php#L81